addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -0,0 +1,216 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+const uuid = require('uuid/v1');
+const { runAndCatch } = require('@aws-ee/base-services/lib/helpers/utils');
+const Service = require('@aws-ee/base-services-container/lib/service');
+const createSchema = require('../schema/create-egress-store.json');
+const {
+  getStatementParamsFn,
+  listStatementParamsFn,
+  putStatementParamsFn,
+  updateS3BucketPolicy,
+  addAccountToStatement,
+  getRevisedS3Statements,
+} = require('../helpers/utils');
+
+const settingKeys = {
+  tableName: 'dbEgressStore',
+  enableEgressStore: 'enableEgressStore',
+  egressStoreBucketName: 'egressStoreBucketName',
+  egressStoreKmsKeyAliasArn: 'egressStoreKmsKeyAliasArn',
+};
+class DataEgressService extends Service {
+  constructor() {
+    super();
+    this.dependency([
+      'aws',
+      'jsonSchemaValidationService',
+      'dbService',
+      'auditWriterService',
+      's3Service',
+      'lockService',
+      'environmentScService',
+    ]);
+  }
+
+  async init() {
+    await super.init();
+    const [dbService] = await this.service(['dbService']);
+    const table = this.settings.get(settingKeys.tableName);
+
+    this._getter = () => dbService.helper.getter().table(table);
+    this._query = () => dbService.helper.query().table(table);
+    this._updater = () => dbService.helper.updater().table(table);
+    this._deleter = () => dbService.helper.deleter().table(table);
+    this._scanner = () => dbService.helper.scanner().table(table);
+  }
+
+  async createEgressStore(requestContext, environment) {
+    const enableEgressStore = this.settings.get(settingKeys.enableEgressStore);
+
+    if (!enableEgressStore) {
+      throw this.boom.forbidden('Unable to create Egress store since this feature is disabled', true);
+    }
+
+    const [validationService, s3Service] = await this.service(['jsonSchemaValidationService', 's3Service']);
+    await validationService.ensureValid(environment, createSchema);
+
+    const bucketName = this.settings.get(settingKeys.egressStoreBucketName);
+    const folderName = `${environment.id}/`;
+
+    try {
+      s3Service.createFolder(bucketName, folderName);
```

Comment:
Add a test for when createFolder fails?

Reply:
Tests added in commit: c0dfd0b9e97060702c0940a873a3f18d29941584